### PR TITLE
feat(transport/requests): RequestsHTTPTransport execute func add extra_args param

### PR DESCRIPTION
**Pull request by ma-pony**
RequestsHTTPTransport execute function support input headers.
like this
```
result = client.execute(
    query, extra_args={"headers": {"authorization": "xxx-123"}}
)
```

### CODE REVIEW

**Comment by leszekhanusz** [comment]
Are you aware that you can already set headers using `RequestHTTPTransport` using the `headers` argument of the transport ?

File: gql/transport/requests.py [hunk 1]
```diff
@@ -145,7 +147,8 @@ def execute(  # type: ignore
 
         # Pass kwargs to requests post method
         post_args.update(self.kwargs)
-
+        if extra_args:
+            post_args.update(extra_args)
```
Comment (leszekhanusz) [inline gql/transport/requests.py:151]: Please add two newlines, one before the if block and one after

**Review by leszekhanusz — CHANGES_REQUESTED** [review]
LGTM except some minor cosmetic requests (newlines)

**Comment by ma-pony** [comment]
> Are you aware that you can already set headers with `RequestHTTPTransport` using the `headers` argument of the transport ?

yes, I know it.
but my case like 
```
transport = RequestTransport(url=url)
client = Client(transport=transport)

user_a_token = client.execute(user_a_login_gql)
user_a_book = client.execute(user_a_book_gql, extra_args={"headers": {"authorization": "user_a_token"}})

user_b_token = client.execute(user_b_login_gql)
user_b_book = client.execute(user_b_book_gql, extra_args={"headers": {"authorization": "user_b_token"}})

...

```

File: gql/transport/requests.py (hunk 1 above)
Comment (ma-pony) [reply gql/transport/requests.py:153]: > Please add two newlines, one before the if block and one after
fixed

**Review by leszekhanusz — APPROVED** [review]

**Pull request merged**